### ISSUE TITLE
Oversized record error from TLS when using proxy protocol

### ISSUE THREAD

**Issue by robert-gomes** [hashicorp/vault#7196]
**Describe the bug**
When proxy protocol is used by a load balancer (Haproxy in this case) and   proxy_protocol_behaviour is set to "use_always". Vault show the following error 
`[INFO]  http: TLS handshake error from client_ip:50214: tls: oversized record received with length 22617`
This error can also be reproduced consistently by sending the following curl request
`curl -v --haproxy-protocol https://vault:8200/v1/sys/health`

**To Reproduce**
1. Run vault with consul backend and   proxy_protocol_behaviour = "use_always"
2. Use Haproxy as loadbalancer and set send-proxy in the server segment
2. Or use curl to emulate the proxy protocol (`curl -v --haproxy-protocol https://vault:8200/v1/sys/health`) same result
3. Client is unable to communicate via proxy and the aforementioned error message can be seen in the vault logs

**Expected behavior**
Vault would work normally as expected the client's IP (instead of loadbalancer IP) would be visible in the vault audit logs

**Environment:**
* Vault Server Version (retrieve with `vault status`): 1.1.3
* Vault CLI Version (retrieve with `vault version`): 1.1.3
* Server Operating System/Architecture: Centos 7

Vault server configuration file(s):

```hcl
backend "consul" {
  address = "consul:8500"
  advertise_addr = "http://127.0.0.1:8200"
  path = "vault"
  scheme = "http"
}

listener "tcp" {
  address = "0.0.0.0:8200"
  cluster_address = "0.0.0.0:8201"
  tls_cert_file = "/config/certs/cert.cer"
  tls_key_file = "/config/certs/pkey.key"
  proxy_protocol_behaviour = "use_always"
  proxy_protocol_authorized_addrs = "0.0.0.0/0"
}

telemetry {
  dogstatsd_addr   = "localhost:8125"
  disable_hostname = true
}
disable_mlock = true
api_addr = "https://vault:8200"
cluster_addr = "https://vault:8201"
ui=true
```

**Additional context**
The original use case was to be able to use Haproxy as loadbalancer and be able to view client's IP in the audit log without terminating SSL at the load balancer. Hence using Haproxy in TCP mode and utilizing proxy protocol.

**Comment by chrishoffman** [comment]
What version of the proxy protocol are you using?  I believe Vault only supports v1.

**Comment by robert-gomes** [comment]
Proxy protocol v1 is used here.

**Comment by michelvocks** [comment]
Hi @robert-gomes!

Thank you for your bug report and sorry for the late reply. 
I've tried to reproduce your mentioned bug but I was unable to do so.
Is there anything else that you've configured/could be important to reproduce this?

Cheers,
Michel

**Comment by catsby** [comment]
Hello! We haven't heard back regarding this issue in over 30 days. In an effort to try and keep our GitHub issues current, we'll be closing this issue in approximately 7 days if we do not hear back regarding this issue. Please let us know if you can still reproduce this issue, and if there is any more information you could share, otherwise we'll be closing this issue around January 30th, 2020

**Comment by catsby** [comment]
Closing for now. Let us know if you have any more information!

**Comment by robert-gomes** [comment]
Sorry about the late reply, our focused shifted to different issues as we had implemented a workaround. This recently came up again while we tried to implement resource quotas (similar to this issue https://github.com/hashicorp/vault/issues/9651).  Upon some further investigation the issue seems to be with our certificates specifically. 
I was able to get working proxy-protocol setup via self signed certificates (https://github.com/robert-gomes/vault-haproxy-docker).
